RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -196,16 +282,58 @@ void
 EvtPlaneProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 {
   using namespace edm;
+  using namespace std;
   using namespace reco;
 
-  int vs_sell;
-  float vzr_sell;
+  if( loadDB_ && (hiWatcher.check(iSetup) || hirpWatcher.check(iSetup)) ) {
+    //
+    //Get Size of Centrality Table
+    //
+    edm::ESHandle<CentralityTable> centDB_;
+    iSetup.get<HeavyIonRcd>().get(centralityLabel_,centDB_);
+    nCentBins_ = centDB_->m_table.size();
+    for(int i = 0; i<NumEPNames; i++) {
+      flat[i]->setCaloCentRefBins(-1,-1);
+      if(caloCentRef_>0) {
+	int minbin = (caloCentRef_-caloCentRefWidth_/2.)*nCentBins_/100.;
+	int maxbin = (caloCentRef_+caloCentRefWidth_/2.)*nCentBins_/100.;
+	minbin/=CentBinCompression_;
+	maxbin/=CentBinCompression_;
+	if(minbin>0 && maxbin>=minbin) {
+	  if(EPDet[i]==HF || EPDet[i]==Castor) flat[i]->setCaloCentRefBins(minbin,maxbin);
+	}
+      }
+    }
+
+    //
+    //Get flattening parameter file.  
+    //
+    edm::ESHandle<RPFlatParams> flatparmsDB_;
+    iSetup.get<HeavyIonRPRcd>().get(flatparmsDB_);
+    LoadEPDB db(flatparmsDB_,flat);
+    if(!db.IsSuccess()) {
+      loadDB_ = kFALSE;
+    }
+
+  } //rp record change
```

Comment:
In order to be safe in terms of event-by-event order, I suggest to use explicitly ``` const HiEvtPlaneFlatten* iFlatConst = flat[i]``` in all remaining parts of the ::produce method. 
With the way the code is written now, it's possible to pass information between events.